packages/vat-data/README.md

```diff
@@ -0,0 +1,35 @@
+# Agoric Vat Data
+
+This package provides access to the Vat Data facility.
+
+## Overview
+
+A [**vat**](https://docs.agoric.com/glossary/#vat) is a unit of isolation. Objects and functions in a JavaScript vat can communicate synchronously with one another. Vats and their contents can communicate with other vats and their objects and functions, but can only communicate asynchronously.
+
+This package provides functions that can persist data in different **zones**:
+
+- heap (holds memory, lost on vat termination)
+- virtual (can be paged out, lost on vat termination)
+- durable (can be paged out, can persist through vat termination and restart)
+
+The kinds defined by the `prepare*` functions (e.g. `prepareExo`) must be available before their data is accessed. Understanding this life cycle requires a couple more concepts:
+
+<!-- TODO include these terms in the glossary -->
+
+A **turn** is everything that happens between an empty stack and the next empty stack, for a given vat.
+
+A **crank** is everything that happens from an empty stack and promise queue to the next empty stack and promise queue.
+
+When a vat is restarted, all the kinds must be defined in the first crank. Consider if some restoration happened in a second crank. If the restart had to wait for external deliveries, the vat would need to somehow enter a suspended state where no other deliveries than the ones needed for completion of start. And that would leave side effects experienced by other vats, so the restart/upgrade could never be fully backed out of it fails.
+
+## Tips
+
+### Synchronous makers
+
+The durable kind maker functions are synchronous. When converting a maker that is async, you'll have to ensure that all necessary data is already available and need not be awaited in the vat's `prepare`.
+
+The reason for this constraint is that _all prepares happen in the first crank_ of the event loop.
+
+For any of the exo instances, remote vats may hold a capability to them. Once the successor vat-incarnation comes online, a message may arrive for any of these instances. For the instance to know how to react to the incoming message, it needs to know the code that defines its behavior. All it got from its ancestor vat-incarnation was its data, not its code.
+
+The successor vat-incarnation must give all outstanding exos their behaviors during the first crank, because that is guaranteed to happen before this vat-incarnation receives any messages sent to those exos.
```

Comment:
```suggestion
Once the successor vat-incarnation comes online, a message may arrive for any exo instance that a previous vat-incarnation exported to another vat. And for that instance to properly react to such an incoming message, implementation code must have already been defined for its Kind. Data persists across vat-incarnations, but code does not.

The successor vat-incarnation must give all outstanding exos their behaviors during the first crank, because that is guaranteed to happen before they receive any messages.
```